optaplanner-core/src/test/java/org/optaplanner/core/impl/domain/common/accessor/lambda/PropertySetterAccessorTest.java

```diff
@@ -0,0 +1,126 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.domain.common.accessor.lambda;
+
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.optaplanner.core.impl.domain.common.ReflectionHelper;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+@RunWith(Parameterized.class)
+public class PropertySetterAccessorTest {
+
+    private final String propertyName;
+    private final Object value;
+
+    public PropertySetterAccessorTest(String propertyName, Object value) {
+        this.propertyName = propertyName;
+        this.value = value;
+    }
+
+    @Parameterized.Parameters(name = "{0}")
+    public static Collection<Object[]> data() {
+        Collection<Object[]> params = new ArrayList<>();
+        params.add(new Object[]{"integerProperty", 5});
+        params.add(new Object[]{"shortProperty", (short) 5});
+        params.add(new Object[]{"byteProperty", (byte) 5});
+        params.add(new Object[]{"booleanProperty", true});
+        params.add(new Object[]{"floatProperty", 5.5f});
+        params.add(new Object[]{"doubleProperty", 5.5});
+        params.add(new Object[]{"objectProperty", "test"});
+        params.add(new Object[]{"charProperty", 'a'});
+        params.add(new Object[]{"arrayProperty", new char[]{'a', 'b'}});
+
+        return params;
+    }
+
+    @Test
+    public void testSetterAccessors() throws NoSuchFieldException, IllegalAccessException {
+        Class<?> declaringClass = MockClassWithSetters.class;
+        Field field = declaringClass.getDeclaredField(propertyName);
+        Class<?> propertyType = field.getType();
+
+        MethodHandles.Lookup lookup = MethodHandles.lookup();
+        Method setterMethod = ReflectionHelper.getSetterMethod(declaringClass, propertyType, propertyName);
+        PropertySetterAccessor<?> propertySetterAccessor =
+                PropertySetterAccessor.create(setterMethod, propertyType, lookup);
+        assertNotNull(propertySetterAccessor);
+
+        MockClassWithSetters mockBean = new MockClassWithSetters();
+        propertySetterAccessor.executeSetter(mockBean, value);
+
+        assertEquals(value, field.get(mockBean));
+    }
+
+    static class MockClassWithSetters {
+
+        int integerProperty;
+        boolean booleanProperty;
+        String objectProperty;
+        double doubleProperty;
+        float floatProperty;
+        short shortProperty;
+        char charProperty;
+        byte byteProperty;
+        char[] arrayProperty;
```

Comment:
let's make these fields private

Reply:
I tend to rather keep it as is, if you don't mind; changing to private would require the test to call setAccessible(true).